climate/vicare.py

```diff
@@ -149,6 +168,34 @@ def turn_away_mode_off(self):
         self.schedule_update_ha_state()
         self._api.setMode("normal")
 
+    def set_hold_mode(self, hold_mode):
+        if hold_mode in ["away", "home"]:
+            active_mode = self._api.getActiveMode()
+            self.schedule_update_ha_state()
+            if hold_mode == "away":
+                success = self._api.setMode("forcedReduced")
+            else:
+                hold_mode == "home"
+                success = self._api.setMode("forcedNormal")
+            # PyViCare currently returns 'None' on no error, checking for both for future changes
+            if success["error"] is None or success["error"] == 'None':
+                self._hold = hold_mode
+                if active_mode in ["standby", "dhw", "dhwAndHeating", "active"]:
+                    self._pre_hold = active_mode
+            else:
+                _LOGGER.error("Failed to set hold mode on ViCare API to %s with status code %s and error %s", hold_mode, success["statusCode"], success["errror"])
+        elif hold_mode == "off":
+            if self._pre_hold is not None:
+                success = self._api.setMode(self._pre_hold)
+            else:
+                _LOGGER.info("No stored pre hold mode found - switching to hot water and heating")
+                success = self._api.setMode("dhwAndHeating")
```

Comment:
On the other hand this might be better than returning to forceReduced or forcedNormal.